deployer/src/deployer/search/models.py

```diff
@@ -76,6 +76,24 @@
     replacement="_$1_",
 )
 
+special_charater_name_char_filter = char_filter(
+    "special_charater_name_char_filter",
+    type="pattern_replace",
+    # In Java, this matches things like `yield*`, `Function*` and `data-*`.
+    # But not that it will also match things like: `x*y` or `*emphasis*` which
+    # is a "risk" worth accepting because event `*emphasis*` would get converted
+    # to `emphasisstar` which is more accurate than letting the standard tokenizer
+    # turn it into `emphasis` alone.
+    pattern="([\\w]+)-?\\*",
```

Comment:
Nit. It seems this could be simplified to:
```suggestion
    pattern="(\\w+)-?\\*",
```

Reply:
Yes! I think it's because I used to have `[a-z]` in there but that implies all lowercase. And `Function*` is a real word, so I later changed it to `\w` and forgot to remove the `[` and `]`. 
Good catch!
Also, `\w` means `[a-zA-Z_0-9]` which I'm not sure is right. But it's hard to tell. 
Currently, based on looking at [every single en-US document `title`](https://github.com/mdn/yari/issues/3113#issue-821248846) I think the regex is sane. That `_` and `0-9` stuff might make a difference within the texts of the body but as long as the first result is what people expect, I'm not too worried.